### PR TITLE
Allow access to real alsa hardware by systemd

### PR DESCRIPTION
**Pull request by Whisprin**
Add `DeviceAllow=char-alsa rw` and disable `PrivateDevices` to allow access to alsa hardware

### CODE REVIEW

File: contrib/airsonic.service [hunk 1]
```diff
@@ -24,8 +24,9 @@ Group=airsonic
 # See https://www.freedesktop.org/software/systemd/man/systemd.exec.html
 # for details
 DevicePolicy=closed
+DeviceAllow=char-alsa rw
 NoNewPrivileges=yes
-PrivateDevices=yes
+#PrivateDevices=yes
```
Comment (almet) [inline contrib/airsonic.service]: If we were to remove this, should we directly remove it, rather than commenting-it-out?

**Comment by almet** [comment]
For whoever is interested, here is the systemd doc about private devices: https://www.freedesktop.org/software/systemd/man/systemd.exec.html#PrivateDevices=

File: contrib/airsonic.service (hunk 1 above)
Comment (Whisprin) [reply contrib/airsonic.service]: Maybe add a comment stating that this is required for Jukebox only, similar to line 46-48: #ProtectHome=true

**Comment by Whisprin** [comment]
Opened an issue for further discussion. With the changes proposed here I'm able to select an audio device from the Admin UI. However, it's still not possible to play music.

File: contrib/airsonic.service (hunk 1 above)
Comment (muff1nman) [reply contrib/airsonic.service]: Yes a comment would be good and moved into a separate section like `ProtectHome`

**Comment by fxthomas** [comment]
I think we should:
* Setup the systemd unit for all features by default
* Comment on why these restrictions were lifted (so that interested people can re-enable them if they don't use the associated feature)

Also, what do you mean by "jukebox is a bit crashy"? Is this something that could be split in a separate issue, or is that related to the device restrictions?

**Comment by Whisprin** [comment]
> Also, what do you mean by "jukebox is a bit crashy"? Is this something that could be split in a separate issue, or is that related to the device restrictions?

Yes, propably something that would go into a separate issue. I'm trying to make it reproducible and collect logs. In short: I use DSub to connect to the jukebox, music is playing, switching songs works - however, when I disconnect DSub and reconnect later: nothing is playing anymore
The only way to recover from this state is to restart Airsonic

File: contrib/airsonic.service (hunk 1 above)
Comment (muff1nman) [reply contrib/airsonic.service]: @Whisprin can you update your PR?

Comment (Whisprin) [reply contrib/airsonic.service]: @muff1nman Updated the PR accordingly, sorry for the delay

**Pull request merged**